src/main/java/mate/academy/bookstore/repository/book/BookSpecificationBuilderImpl.java

```diff
@@ -0,0 +1,40 @@
+package mate.academy.bookstore.repository.book;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstore.dto.BookSearchParametersDto;
+import mate.academy.bookstore.model.Book;
+import mate.academy.bookstore.repository.SpecificationProviderManager;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationBuilderImpl implements BookSpecificationBuilder {
+    private static final String TITLE_KEY = "title";
+    private static final String AUTHOR_KEY = "author";
+    private static final String PRICE_KEY = "price";
+    private final SpecificationProviderManager<Book> specificationProviderManager;
+
+    @Override
+    public Specification<Book> build(BookSearchParametersDto bookSearchParametersDto) {
```

Comment:
OPTIONAL: Actually this method violates open-closed principle. It is possible to implement strategy here or think how it could be without numbers of if statements